doc/source/reference/arrays.classes.rst

```diff
@@ -665,9 +665,9 @@ objects as inputs and returns an iterator that returns tuples
 providing each of the input sequence elements in the broadcasted
 result.
 
->>> for val in np.broadcast([[1,0],[2,3]],[0,1]):
+>>> for val in np.broadcast([[1, 0], [2, 3]], [0, 1]):
 ...     print(val)
-(1, 0)
-(0, 1)
-(2, 0)
-(3, 1)
+(np.int64(1), np.int64(0))
+(np.int64(0), np.int64(1))
+(np.int64(2), np.int64(0))
+(np.int64(3), np.int64(1))
```

Comment:
Didn't the text of NEP 51 say:
> The NEP does not propose to change how these scalars print – only
their representation (``__repr__``) will be changed.

Reply:
Tuples print the `repr` of their content.

There already is `.item()` to convert to the python scalar where applicable.  The only problem is longdouble, which should be rare.

About changing how this is backed: I am happy for someone to think about it, but it is neither this, nor will I seriously consider breaking ABI.  Propose it as an API break and I will be interested.